servatrice/src/smtpclient.cpp

```diff
@@ -82,6 +82,54 @@ bool SmtpClient::enqueueActivationTokenMail(const QString &nickname, const QStri
     return true;
 }
 
+bool SmtpClient::enqueueForgotPasswordTokenMail(const QString &nickname, const QString &recipient, const QString &token)
+{
+	QString email = settingsCache->value("forgotpassword/email", "").toString();
+	QString name = settingsCache->value("forgotpassword/name", "").toString();
+	QString subject = settingsCache->value("forgotpassword/subject", "").toString();
+	QString body = settingsCache->value("forgotpassword/body", "").toString();
+
+	if (email.isEmpty())
+	{
+		qDebug() << "[MAIL] Missing sender email in configuration";
+		return false;
+	}
+
+	if (subject.isEmpty())
+	{
+		qDebug() << "[MAIL] Missing subject field in configuration";
+		return false;
+	}
+
+	if (body.isEmpty())
+	{
+		qDebug() << "[MAIL] Missing body field in configuration";
+		return false;
+	}
+
+	if (recipient.isEmpty())
+	{
+		qDebug() << "[MAIL] Missing recipient field for user " << nickname;
+		return false;
+	}
+
+	if (token.isEmpty())
+	{
+		qDebug() << "[MAIL] Missing token field for user " << nickname;
+		return false;
+	}
+
+	QxtMailMessage message;
+	message.setSender(name + " <" + email + ">");
+	message.addRecipient(recipient);
+	message.setSubject(subject);
+	message.setBody(body.replace("%username", nickname).replace("%token", token));
+
+	int id = smtp->send(message);
```

Comment:
For starters, this works the exact same way the registration works (so not sure why we want it to work differently), the other thing to mention is all this does is add the email to the que that is later processed at the next timer countdown.  I could look to see if the function in the smtp client returns something different or a value to check but I'm not sure if that should really hold the process up.  The end result would be the user looks for an email that simply never comes and then contacts the server operator for help.  Inconvenience sure, but its not really any different than the current behavior when a user goes to register.